firmware/mw_dual_motor_torque_ctrl/src/dual_motor_torque_ctrl.c

```diff
@@ -1171,34 +1196,53 @@ inline void generic_motor_ISR(const HAL_MtrSelect_e mtrNum)
 			speed_pu = STPOSCONV_getVelocity(st_obj[mtrNum].posConvHandle);
 		}
 		else
-		{  // the alignment procedure is in effect
-
-			// force motor angle and speed to 0
-			angle_pu[mtrNum] = _IQ(0.0);
-			speed_pu = _IQ(0.0);
-
-			// set D-axis current to Rs estimation current
-			gIdq_ref_pu[mtrNum].value[0] = _IQmpy(
+		{
+			// The alignment procedure is in effect.
+			// For the first gCalibrationWrampupTime steps, wramp up towards the
+			// calibration current slowly. Then, hold the current for
+			// the duration of gCalibrationHoldTime steps.
+			if(gAlignCount[mtrNum]++ < gCalibrationWrampupTime) {
+				angle_pu[mtrNum] = _IQ(0.0);
+				speed_pu = _IQ(0.0);
+
+				// Set D-axis current to slowly increasing estimation current.
+				_iq idq_ref_pu_max = _IQmpy(
+						_IQ(gUserParams[mtrNum].maxCurrent_resEst),
+						gCurrent_A_to_pu_sf[mtrNum]);
```

Comment:
```suggestion
				_iq idq_ref_pu_max = gUserParams[mtrNum].maxCurrent_resEst
				                     * gCurrent_A_to_pu_sf[mtrNum];
```
As this was already like this in the old code, I might be missing something here, but I think multiplication of a Q-number with a "normal" number can just be done normally.